apps/token-manager/app/src/hooks.js

```diff
@@ -0,0 +1,82 @@
+import { useRef, useEffect, useCallback, useState } from 'react'
+
+export function useClickOutside(cb, existingRef) {
+  const ref = existingRef || useRef()
+  const handleClick = useCallback(
+    e => {
+      if (!ref.current.contains(e.target)) {
+        cb()
+      }
+    },
+    [cb, ref]
+  )
+
+  useEffect(() => {
+    document.addEventListener('click', handleClick, true)
+    return () => {
+      document.removeEventListener('click', handleClick, true)
+    }
+  }, [handleClick])
+
+  return { ref }
+}
+
+export function useOnBlur(cb, existingRef) {
+  const ref = existingRef || useRef()
+  const handleBlur = useCallback(
+    e => {
+      if (!ref.current.contains(e.relatedTarget)) {
+        cb()
+      }
+    },
+    [cb, ref]
+  )
+
+  return { ref, handleBlur }
+}
+
+export function useArrowKeysFocus(query, containerRef = useRef()) {
+  const [index, setIndex] = useState(-1)
+
+  const reset = () => setIndex(-1)
+  const cycleFocus = useCallback(
+    change => {
+      const elements = document.querySelectorAll(query)
+      let next = index + change
+      if (next > elements.length - 1) {
+        next = 0
+      }
+      if (next < 0) {
+        next = elements.length - 1
+      }
+      if (!elements[next]) {
+        next = -1
+      }
+      setIndex(next)
+    },
+    [index]
+  )
+  const handleKeyDown = useCallback(
+    ({ keyCode }) =>
+      keyCode === 38 ? cycleFocus(-1) : keyCode === 40 ? cycleFocus(1) : null,
```

Comment:
What do you think about switching the "magic numbers" with constants:
```suggestion
      keyCode === KEYCODE_UP ? cycleFocus(-1) : keyCode === KEYCODE_DOWN ? cycleFocus(1) : null,
```